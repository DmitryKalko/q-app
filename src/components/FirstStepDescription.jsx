import React from "react";
import firstStep from "./svg/first-step.svg";

const FirstStepDescription = ({ lightTheme }) => {
  return (
    <div className="step">
      <p style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}>Hey!</p>
      <p style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}>
        What is your name?
      </p>
      <img src={firstStep} alt="firstStep" />
      <p>dh</p>
    </div>
  );
};
export default FirstStepDescription;
