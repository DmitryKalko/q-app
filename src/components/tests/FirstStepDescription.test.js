import { shallow } from "enzyme";
import React from "react";
import FirstStepDescription from "../FirstStepDescription";

describe('about work FirstStepDescription component', () => {
    it('should render FirstStepDescription component', () => {
        const component = shallow(<FirstStepDescription/>)
        expect(component).toMatchSnapshot();
    });
});

// тестируя снапшотами разметку ты можешь проверять правильность логики
// например пропсы, ты получил пропс и хочешь что-то отрендорить(соответственно у тебя есть снимок как это должно выглядеть)
// кто-то что-то поменял и пропсы не приходят - соответственно ошибка в тесте, как как разметка выглядит уже подругому