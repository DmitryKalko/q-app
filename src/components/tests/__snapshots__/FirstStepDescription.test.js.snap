// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about work FirstStepDescription component should render FirstStepDescription component 1`] = `
<div
  className="step"
>
  <p
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
  >
    Hey!
  </p>
  <p
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
  >
    What is your name?
  </p>
  <img
    alt="firstStep"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <p>
    dh
  </p>
</div>
`;
